vispy/color/colormap.py

```diff
@@ -116,44 +120,82 @@ def step(colors, x, controls=None):
 
 
 # GLSL interpolation functions.
-def _glsl_mix(controls=None):
+def _glsl_mix(controls=None, colors=None, texture_map_data=None):
```

Comment:
The function signature for `_glsl_mix` and `_glsl_step` seem a little odd to me since they have assertion checks on some of the arguments not being None or being a certain way, but they are still keyword arguments. Seems like they should be positional arguments, but I suppose having them the way they are probably matches up better with other user methods and there is no point in breaking things for older user code that may be depending on this...I think.